File: code/framework-comparison/markup-renderer/markup_renderer.nim
import strutils, re

proc renderMarkup(text: string): string =
  let markupTags = @[
    r"\[title\](.*?)\[/title\]" -> "\x1B[33;1m\\1\x1B[0m",    # Yellow bold text
    r"\[emphasis\](.*?)\[/emphasis\]" -> "\x1B[31;1m\\1\x1B[0m" # Red bold text
  ]

  var result = text
  for (tag, replacement) in markupTags:
    result = result.re.replace(tag, replacement)
  return result

proc main() =
  let inputFilename = "input.txt"
  try:
    let content = readFile(inputFilename)
    let renderedContent = renderMarkup(content)
    echo renderedContent
  except IOError:
    echo "File '", inputFilename, "' not found."

when isMainModule:
  main()

File: code/framework-comparison/markup-renderer/markup-renderer.js
const fs = require('fs');

const markupTags = [
    { tag: /\[title\](.*?)\[\/title\]/g, replacement: '\x1b[33;1m$1\x1b[0m' },    // Yellow bold text
    { tag: /\[emphasis\](.*?)\[\/emphasis\]/g, replacement: '\x1b[31;1m$1\x1b[0m' } // Red bold text
];

function renderMarkup(text) {
    for (const tag of markupTags) {
        text = text.replace(tag.tag, tag.replacement);
    }
    return text;
}

const inputFilename = 'input.txt';

fs.readFile(inputFilename, 'utf8', (err, content) => {
    if (err) {
        console.log(`File '${inputFilename}' not found.`);
        return;
    }
    
    const renderedContent = renderMarkup(content);
    console.log(renderedContent);
});
